cmd/cb-event-forwarder/amqp.go

```diff
@@ -66,6 +67,7 @@ func NewConsumer(amqpURI, queueName, ctag string, bindToRawExchange bool,
 		dialer:            dialer,
 		amqpURI:           amqpURI,
 		queueName:         queueName,
+		connectionErrors:  make(chan *amqp.Error),
```

Comment:
Would you please explain what this change does?

Reply:
`c.conn.NotifyClose(c.connectionErrors)` is called in Conumser::Connect() but because c.connecitonErrors was left as null by this factory-method , the amqp connection had nobody to notify - this caused the regression where the forwarder failed to realize the broker had hung up/forced the channel down. 

https://godoc.org/github.com/streadway/amqp#Channel.NotifyClose

for deets

Reply:
Ah, ok, channel ops on a nil/null channel block forever - only close() causes a runtime panic so no error being observed makes sense. Sneaky sneaky.

Reply:
Tl;DR - NewConsumer() should properly construct a consumer, it wasn't - the `connectionErrors` channel as left nil causing strangeness at runtime when the broker signals that the channel is being torn-down.